src/wallet/gtest/test_wallet.cpp

```diff
@@ -781,39 +849,41 @@ TEST(wallet_tests, CachedWitnessesCleanIndex) {
         ZCIncrementalMerkleTree sproutRiPrevTree {sproutRiTree};
         ZCSaplingIncrementalMerkleTree saplingRiPrevTree {saplingRiTree};
         wallet.IncrementNoteWitnesses(&(indices[i]), &(blocks[i]), sproutRiTree, saplingRiTree);
-        witnesses.clear();
-        uint256 anchor;
-        wallet.GetSproutNoteWitnesses(notes, witnesses, anchor);
+
+        auto anchors = GetWitnessesAndAnchors(wallet, sproutNotes, saplingNotes, sproutWitnesses, saplingWitnesses);
         for (size_t j = 0; j < numBlocks; j++) {
-            EXPECT_TRUE((bool) witnesses[j]);
+            EXPECT_TRUE((bool) sproutWitnesses[j]);
+            EXPECT_TRUE((bool) saplingWitnesses[j]);
         }
         // Should equal final anchor because witness cache unaffected
-        EXPECT_EQ(anchors.back(), anchor);
+        EXPECT_EQ(sproutAnchors.back(), anchors.first);
+        EXPECT_EQ(saplingAnchors.back(), anchors.second);
 
         if ((i == 5) || (i == 50)) {
             // Pretend a reorg happened that was recorded in the block files
             {
                 wallet.DecrementNoteWitnesses(&(indices[i]));
-                witnesses.clear();
-                uint256 anchor;
-                wallet.GetSproutNoteWitnesses(notes, witnesses, anchor);
+
+                auto anchors = GetWitnessesAndAnchors(wallet, sproutNotes, saplingNotes, sproutWitnesses, saplingWitnesses);
                 for (size_t j = 0; j < numBlocks; j++) {
-                    EXPECT_TRUE((bool) witnesses[j]);
+                    EXPECT_TRUE((bool) sproutWitnesses[j]);
+                    EXPECT_TRUE((bool) saplingWitnesses[j]);
                 }
                 // Should equal final anchor because witness cache unaffected
-                EXPECT_EQ(anchors.back(), anchor);
+                EXPECT_EQ(sproutAnchors.back(), anchors.first);
+                EXPECT_EQ(saplingAnchors.back(), anchors.second);
             }
 
             {
                 wallet.IncrementNoteWitnesses(&(indices[i]), &(blocks[i]), sproutRiPrevTree, saplingRiPrevTree);
-                witnesses.clear();
-                uint256 anchor;
-                wallet.GetSproutNoteWitnesses(notes, witnesses, anchor);
+
```

Comment:
Why is there not another call to ``GetWitnessesAndAnchors`` here?

Reply:
This is a mistake. Thanks for catching this!